### PR TITLE
Enable Github pages docs deploy (master /docs)

### PR DESCRIPTION
**Pull request by jonathanstallings**
Addresses issue #22 

Taking feedback from PR #56, this changeset is set to serve from `master` branch `/docs`.

Principal changes:
* Cleaned up unneeded README references to IMS specific instructions.
* Removed unneeded docker/S3 deployment files
* Moved documentation source files to separate `docs-src` directory (necessary -- mkdocs needs a separate directory for source from build output directory)
* Updated gitignore file

### CODE REVIEW

**Comment by jonathanstallings** [comment]
TODO: Update `Github Pages` config in repo settings to target `master` branch `/docs` folder after merge.

File: docs-src/Makefile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+deps:
+	pip install awscli mkdocs
```
Comment (zpatrick) [inline docs-src/Makefile]: [nit] I don't we need awscli anymore do we?

File: docs-src/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Layer0 Documentation - Developer Notes
+
+
+## Local Development
+
+- `make deps` to ensure you have `mkdocs` and `awscli` installed.
```
Comment (zpatrick) [inline docs-src/README.md]: *refers to awscli comment above*

**Comment by zpatrick** [comment]
:+1: sweet! looks good

File: docs-src/Makefile (hunk 1 above)
Comment (jonathanstallings) [inline docs-src/Makefile]: True that!

File: docs-src/README.md (hunk 2 above)
Comment (jonathanstallings) [inline docs-src/README.md]: Agreed.

**Pull request merged**